### PR TITLE
Reorganizing & cleanup

### PR DESCRIPTION
**Pull request by EvanBoyle**
In an effort to make this more "library like" and to make it easier to publish a package to npm, I moved things around so that the core code in in `/lib` and the example usage is in `/examples`. Also cleaned up some leftovers.

### CODE REVIEW

File: lib/serverless/pipeline/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as pulumi from "@pulumi/pulumi";
 import * as aws from "@pulumi/aws";
-import { input } from "@pulumi/aws/types/index";
+import { input } from "@pulumi/aws/types";
```
Comment (jmaysrowland) [inline lib/serverless/pipeline/index.ts]: Why do we no longer need to point directly to index?

File: lib/serverless/datawarehouse/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as pulumi from "@pulumi/pulumi";
 import * as aws from "@pulumi/aws";
-import { input } from "@pulumi/aws/types/index";
-import { getS3Location } from "../../utils";
+import { input } from "@pulumi/aws/types";
```
Comment (jmaysrowland) [inline lib/serverless/datawarehouse/index.ts]: Why do we no longer need to point directly to index?

File: lib/serverless/pipeline/index.ts (hunk 1 above)
Comment (EvanBoyle) [reply lib/serverless/pipeline/index.ts]: I ran an automated import updating command when I moved the directory structure around. This was the result 🤷‍♂

Comment (EvanBoyle) [reply lib/serverless/pipeline/index.ts]: But I believe that the two statements are effectively the same.

**Pull request merged**